app/Services/PostService.php

```diff
@@ -39,7 +40,59 @@ public function create($data, $signupId, $transactionId)
         return $post;
     }
 
-    /*
+    /**
+     * Handles all business logic around updating posts.
+     *
+     * @param \Rogue\Models\Signup $signup
+     * @param array $data
+     * @param string $transactionId
+     *
+     * @return \Illuminate\Database\Eloquent\Model $model
+     */
+    public function update($signup, $data, $transactionId)
+    {
+        $this->resolvePostRepository($data['event_type']);
+
+        $post = $this->repository->update($signup, $data);
+
+        // @TODO: This will is only temporary and will be removed!
+        // If this is a signup update, get the most recent post.
+        // If there is a quantity_pending, this is a signup.
+        if ($post->quantity_pending) {
+            $signupId = $post->id;
+            // Find the post with this signup id.
+            $post = Post::where('signup_id', $signupId)->first();
+        }
+
+        // Add new transaction id to header.
+        request()->headers->set('X-Request-ID', $transactionId);
+
+        // Post reportback back to Phoenix, unless told not to.
+        // If request fails, record in failed_jobs table.
+        if (! isset($data['do_not_forward'])) {
+            dispatch(new SendPostToPhoenix($post, isset($data['file'])));
+        }
+
+        return $post;
+    }
+
+    /**
+     * Handles all business logic around updating the posts(s)'s status after being reviewed.
+     *
+     * @param array $data
+     *
+     * @return
+     */
+    public function reviews($data)
+    {
+        $this->resolvePostRepository($data[0]['event_type']);
```

Comment:
@sbsmith86 since this is a nested array and admins have the ability to review multiple posts at once, I am just checking what type it is for the first post in the array. Right now, they will always be photos since that's the only functionality we have. 

Short answer: Yes. Even if there is only one photo being reviewed, it is still in a nested array. 

Something to think about in the future: will admins be able to review photos and videos (or any other type) all at once? If so, this logic will need to change. 

Should I add a @TODO note here regarding this?